include/sycl_ext_complex.hpp

```diff
@@ -1232,6 +1240,436 @@ _SYCL_EXT_CPLX_INLINE_VISIBILITY complex<_Tp> tan(const complex<_Tp> &__x) {
 
 _SYCL_EXT_CPLX_END_NAMESPACE_STD
 
+////////////////////////////////////////////////////////////////////////////////
+// MARRAY IMPLEMENTATION
+////////////////////////////////////////////////////////////////////////////////
+
+_SYCL_BEGIN_NAMESPACE
+
+// marray of complex class specialisation
+template <typename T, std::size_t NumElements>
+class marray<sycl::ext::cplx::complex<T>, NumElements> {
+private:
+  using DataT = sycl::ext::cplx::complex<T>;
+
+public:
+  using value_type = DataT;
+  using reference = DataT &;
+  using const_reference = const DataT &;
+  using iterator = DataT *;
+  using const_iterator = const DataT *;
+
+private:
+  value_type MData[NumElements];
+
+public:
+  constexpr marray() : MData{} {};
+
+  explicit constexpr marray(const DataT &arg) {
+    for (size_t i = 0; i < NumElements; ++i)
+      MData[i] = arg;
+  }
+
+  template <typename... ArgTN>
+  constexpr marray(const ArgTN &...args) : MData{args...} {};
+
+  constexpr marray(const marray<DataT, NumElements> &rhs) = default;
+  constexpr marray(marray<DataT, NumElements> &&rhs) = default;
+
+  // Available only when: NumElements == 1
+  template <typename = typename std::enable_if<NumElements == 1>>
+  operator DataT() const {
+    return MData[0];
+  }
+
+  static constexpr std::size_t size() noexcept { return NumElements; }
+
+  marray<T, NumElements> real() const {
+    sycl::marray<T, NumElements> rtn;
+
+    for (std::size_t i = 0; i < NumElements; ++i) {
+      rtn[i] = MData[i].real();
+    }
+
+    return rtn;
+  }
+
+  marray<T, NumElements> imag() const {
+    sycl::marray<T, NumElements> rtn;
+
+    for (std::size_t i = 0; i < NumElements; ++i) {
+      rtn[i] = MData[i].imag();
+    }
+
+    return rtn;
+  }
+
+  // subscript operator
+  reference operator[](std::size_t index) { return MData[index]; }
```

Comment:
We use `i` in the foor loop on top, I guess we we can use `i` for index everywhere

Reply:
Updated the variable name